Tags and other litlle features added

Flyounet opened the pull request:
hi,

The 6 previous PRs are already included in this one :
- _DraftDir + manage accent for slug()_
- _Prev and Next Post_
- _Toggle status + list posts_
- _Put config in a function_
- _Remove from the main and create functions_
- _New feature : Tags_

Not sure about the way you wanted the tags, but here is an implementation.

Each post could have more than one tag, and each tag links to an index where posts have the same tags.

This PR was the first done (now splitted in more little PRs).

Regards

Flyounet committed:
Baker now creates a Draft dir and generate post for drafted post.

Flyounet committed:
For people like me (who's speaking french), accent are important.
Previously the slug function didn't handled accent :
echo 'Baker est très bien' | tr -cs '[:alnum:]\n' - | tr '[:upper:]' '[:lower:]' | sed 's|^-*||;s|-*$||'
baker-est-tr-s-bien

Now it handles them :
echo 'Baker est très bien' | iconv -f utf8 -t ascii//TRANSLIT | tr -cs '[:alnum:]\n' - | tr '[:upper:]' '[:lower:]' | sed 's|^-*||;s|-*$||'
baker-est-tres-bien

Flyounet committed:
Starting to implement a sort of Prev/Next button for each post.
The idea is when you're on a Post, you can go to the older or the newer.
If the there is no newer or no older, you go to the index.

Flyounet committed:
Merge branch 'draftdir' into prevnextpost

 pour expliquer en quoi cette fusion est

Flyounet committed:
When posts are baked, they now points to older are newer posts.

Flyounet committed:
Baker now list all posts with their status

Flyounet committed:
Baker now is able to easily change the status of a post

Flyounet committed:
Adding help for toggle

Flyounet committed:
Changed tog by toggle in the switch to follow help

Flyounet committed:
Adding info about toggle in the README

Flyounet committed:
Typo error in the README

Flyounet committed:
The actions : post, list and toggle are now functions. It will be easier to do changes if needed.

Flyounet committed:
The actione : bake is now a function post_bake.

Flyounet committed:
Configuration is now in a function. When i'll have finishing to put everything we'll be able to send a config file to baker.

Flyounet committed:
Merge branch 'functionalize' into dev
All actions list, post, bake, toggle are now functions.
Conflicts:
	baker

Flyounet committed:
Adding information to end user

Flyounet committed:
Define the base name for categories

Flyounet committed:
It almost works \o/. Categories are generated.

Flyounet committed:
Categories are well generated in the post now (updated layout and posts).
Added the draft directory to .gitignore

Flyounet committed:
Starting to rename categories by tags. It's the name the original author wanted.

Flyounet committed:
All categories reference should have been removed and replaced with tags.

Flyounet committed:
Indexes for tags are generated.

Flyounet committed:
Merging "Tags" (branch named categories) into Dev. Once tested, could be merge to master.

Flyounet committed:
Small update for the README.

taylorchu commented:
This is nice. Could you break it down into many PRs? I want to review and merge them separately.

Flyounet committed:
New feature : Baker could change the date of a post without editing it.

Flyounet committed:
Starting to work on the modifications

Flyounet committed:
Function die created to send message to stderr and exit
As requested the DRAFT_DIR and OUTPUT_DIR are removed and created in same command
Arguments are now also managed like flags
New feature : the -w flag force the use of this workingDir. With standard config, you could use a different working directory

Flyounet committed:
Adding help for flags. Changing some if..then..fi to [[ ]] &&. Use readlink for the configfile.

Flyounet committed:
Adding feature to rename the post by reparsing the post header.

Flyounet committed:
New feature : Rename a post based on the title/date. If you change anything, baker -r id will rename your post.

Flyounet committed:
New feature embed the Markdown.pl directly in the shell script. Now, only one file to manage.

Flyounet committed:
Bug corrected. I called bake function instead of post_bake function.

Flyounet committed:
When printing the list now also include tags

Flyounet committed:
As the Markdown.pl is embedded. No need of the SCRIPT_DIR var.

Flyounet committed:
New header: update : Nobody's perfect, it could happen to amend a post. to warn user, you could add the 'update' header with datetime.

Flyounet committed:
the 'update' header (with a date) inform reader about update in post.
Merge branch 'unmerged-postupdated' into dev

Flyounet committed:
Baker is now first checking if it misses any commands.

Flyounet committed:
Checking first missing commands.
Merge branch 'checkBinaries' into dev

Flyounet committed:
Forget to remove an unused line

Flyounet committed:
Removing unused line. Check if binaries are present to work correctly.
Merge branch 'dev'

Flyounet committed:
Added a new black flavour, and informed changed the README.md to inform users.

Flyounet committed:
Oups... I forget the black flavour. Added layout, public and config.

Flyounet committed:
Typo changes

Flyounet committed:
For new user, the script check if commands are unknown.
Merge branch 'dev'

Flyounet commented:
Since the first PRs, the following things have been added :
- _Change the date of a post (and consequently) its name_ (new feature)
- _printing list also include tags_ (enhancement)
- _Re-parsing the headers title and date to regenerate the post name_ (new feature)
- _New header `update` for when you update a post_ (enhancement)
- _check if commands are missing before starting to work_ (enhancement)
- _A second layout + css_

Flyounet committed:
This new subversion release will include the source of Markdown.pl directly. To only have one file to manage.
Merge branch 'embeddedMarkdown' into branch-1.1.0

Flyounet committed:
Implenting RSS feed for Baker. It will be a sort of index.md with XML tags for RSS.

Flyounet committed:
Repacing FontAwesome 4.4.0 by 4.5.0 in the layout_black/public_black template

Flyounet committed:
After some test. It seems to work properly. The RSS seems to work also.

Flyounet committed:
Corrections: in usage() text for -r corrected. in checkBinaries() check now for column. in post_bake() changed the text for stdout.

Flyounet committed:
Merge branch 'branch-1.0.0-rc3' into dev. Cosmetic corrections mainly.

Flyounet committed:
Merge branch 'dev'. Cosmetic changes. Fontawesome upgraded 4.5.0 for black_layout.

Flyounet commented:
The above commits add :
- the new version of fonteawesome for the second layout
- a little correction for the help of the `-r`
- the verification of `column` and change of text when baking.
- the `1.0.0-rc3`, if no errors found i suggest to publish a `1.0.0` this week end.

The `1.0.1` will include `RSS feed` and `1.0.2` will include `Atom feed`.

Flyounet committed:
Merge branch 'newfeat-rss' into branch-1.0.1. branch-1.0.1 should include RSS Feed.

Conflicts:
	README.md

Flyounet committed:
Update usage() text

Flyounet committed:
Adding an atom feed file and start to work on it.

Flyounet committed:
Starting the default Baker config file loader.

Flyounet committed:
New feature: A default config file (baker.conf) is loaded if found. Use full when you use -w path and you don't want to also use -f configfile.

Flyounet committed:
Merge branch 'defaultBakerConf' into branch-1.0.1. Fixe #2 : Default config file loaded

Flyounet committed:
Fix #9. The </channel> was not in the correct place, and date are now RFC 822 compliant.
The RSS feed is now valid and had passed the http://feedvalidator.org/ test.

Flyounet committed:
Changed rss.md to describe the generator

Flyounet committed:
Updated the README.md.

Flyounet committed:
Creating soe md file to generate a big list after.

Flyounet committed:
New fetaure: adding the summary in the post + README changed.

Flyounet committed:
Changing black_layout + README change

Flyounet committed:
Finished to change the README and an argument to manage the PRINT_ALL_SUMMARY.

Flyounet committed:
Fix #11. Adding summary in the post. Merge branch 'summaryPrint' into branch-1.0.2

Flyounet committed:
Fix #3. Baker has an Atom feed

Flyounet committed:
Adding the RSS link on the original index layout.

Flyounet committed:
Adding files

Flyounet committed:
Pushing RSS. Merge branch 'branch-1.0.1' into dev

Flyounet committed:
Fix #4 - Publishing v1.0.1 - Merge branch 'dev'

Flyounet commented:
As describe in the previous comment, I published a v1.0.0 d5449a3 
And since then, the RSS Feature has been completed in the v1.0.1 45a9e52 (just published)

Flyounet committed:
Adding almost nothing

Flyounet committed:
Generator almost ready

Flyounet committed:
Remove the testing subdir from versionning

Flyounet committed:
Adding a config file for the testing part

Flyounet committed:
Adding a .gitattributes file + a Post generator script + Baker time "Analyzer"

Flyounet committed:
Merge branch 'testingScript' into branch-1.0.2

Flyounet committed:
Cosmetic code change

Flyounet committed:
Unpdating README

Flyounet committed:
Fixes #3 #7 #8 #11
Atom feed added,
Default configuration file,
Summary print in posts,
A little tool to generate blog post to make time statistics
Merge branch 'branch-1.0.2' into dev

Conflicts:
	README.md
	baker

Flyounet committed:
Adding atom.xml link in the layouts.

Flyounet committed:
Fix #12 : Remove the -c and use correctly the -t and -s.
chronoBaker now take arguments to give them to baker.

Flyounet committed:
Permission change

Flyounet committed:
Typo error in README.md

Flyounet committed:
Fix #5 : Delivering v1.0.2 with Atom feed, auto config,...
Merge branch 'dev'

Flyounet commented:
Hi,

I forget to mention, I published a v1.0.2 with Atom Feed, the summary that coul be added in the post (or not), ...

Bye

Flyounet committed:
Adding the md5sum check of the Markdown file.

Flyounet committed:
The Headers are now allowed to have space(s) before and after the header name and the :

Flyounet committed:
Adding a space after the summary.
Adding the sumprint header.

Flyounet committed:
Markdown is now embedded in the shell to have only one file to manage.
Merge branch 'embeddedMarkdown' into testMergeEmbedded

Conflicts:
	baker

Flyounet committed:
Removing the load of Markdown.pl from render_file_case().
Loaded in post_bake to have the variable filled only once.
And also look for the correct bakeer file.

Flyounet committed:
Applying the Merge of Markdown.pl and baker
Merge branch 'testMergeEmbedded' into branch-1.1.0

Flyounet committed:
Adding then -v to print version

Flyounet committed:
Adding git informations

Flyounet committed:
Merge branch 'branch-1.1.0' into dev

Flyounet committed:
Adding a Makefile to install baker

Flyounet committed:
Adding Hidden header/feature + -I to print config

Flyounet committed:
Release v1.1.0 :

- Markdown perl script inluded in baker,
- -v for version

Flyounet committed:
Updating the version

Flyounet committed:
Merge branch 'branch-1.1.1' into dev

Flyounet committed:
Feature update : Adding -e to edit a post.

Flyounet committed:
Readme Update

Flyounet committed:
Preparing release 1.1.1
Updating Licences, as I publish under DSSL & WTFPL (LPRAB).

Flyounet committed:
Release v1.1.1
Including `hidden` header. -I for Infos.

Flyounet committed:
By using -o id or title, baker will open your page with xdg-open

Flyounet committed:
Preparing Release v1.1.2
Merge branch 'openPage' into dev

Flyounet committed:
Release v1.1.2

Flyounet committed:
Preparing v1.1.3
Correct issue with -z flag
Markdown in summary is now converted in index.html.

Flyounet committed:
Preparing v1.1.3
Markdown also replaced in index_tags.html pages.

Flyounet committed:
Preparing v1.1.3
Finally, summary are also transformed when generating a post.

Flyounet committed:
Preparing v1.1.3
Adding color for draft/published and Hidden in list.

Flyounet committed:
Release 1.1.3
Markown parsed for Summary.
Color for list.

Flyounet committed:
Forget a $ for the Format:%h. Thanks to the github support.

Flyounet committed:
Release v1.1.3a
Correction for the Format:%h issue.

Flyounet committed:
Forget to update the version

Flyounet committed:
Importing master change

Flyounet committed:
Transforming the Index into a time line

Flyounet committed:
Updating css

Flyounet committed:
testing

Flyounet committed:
Adding the BAKER_EXTRA_HEADER to add header when creating a post.

Flyounet committed:
Release 1.1.4

Flyounet committed:
Merge branch 'timeline' into Tline

Flyounet committed:
Remove Default unusued BAKER_EXTRA_HEADER

Flyounet committed:
Adding a directory with example
Template modifications

Flyounet committed:
Updating example

Flyounet committed:
Added the -u <id> to Baker, to directly update the update field in a post.
Adding a configuration variable to have a index layout for tags different from the main index.
A GENERATE_INDEX variable is set before generating index (and unset after) and a GENERATE_TAGS is set before doing tags.
Updating the configuration displayed to take into account last variables
Updating the README, the docs seems better.
Adding a config file file to generate the example site with the roiginal template.

Flyounet committed:
Merge branch 'Tline' into dev
Preparing the 1.2 release

Flyounet committed:
Preparing Release v1.2.0

Flyounet committed:
Updating README.

Flyounet committed:
REAMDE update.

Flyounet committed:
Adding a VERSION file

Flyounet committed:
Release v1.2.0

Flyounet committed:
Last commit on Baker.

Flyounet committed:
Very last commit on Baker.
